api/types/mount/mount.go

```diff
@@ -60,6 +61,20 @@ var Propagations = []Propagation{
 	PropagationSlave,
 }
 
+// State represents the consistency requirements of a mount.
+type State string
+
+const (
+	// StateConsistent guarantees bind-mount-like consistency
+	StateConsistent State = "consistent"
```

Comment:
`synchronous` seem to refer more to the mechanism used to accomplish the consistency.

Where as `consistent` refers to the desired "state".

I prefer `consistent` here.